### PR TITLE
Add http host tests v3

### PR DESCRIPTION
**Pull request by jmtaylor90**
fixes from comments in #856

### CODE REVIEW

File: tests/test-bad-http-host-rule-1/test.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+requires:
+  min-version: 7.0.0
+
+  features:
+    - HAVE_LIBJANSSON
+
+command: |
+  ${SRCDIR}/src/suricata --set classification-file="${SRCDIR}/classification.config" --set reference-config-file="${SRCDIR}/reference.config" -l ${OUTPUT_DIR} -c ${TEST_DIR}/suricata.yaml -r ${TEST_DIR}/ -S ${TEST_DIR}/test.rules
```
Comment (catenacyber) [inline tests/test-bad-http-host-rule-1/test.yaml:8]: Why do we need a specific command ?

Comment (jmtaylor90) [reply tests/test-bad-http-host-rule-1/test.yaml:8]: I don't know that it's needed, it's likely just from an existing test case I copied to create these two tests or I just copied the test.yaml from the readme. If it isn' t needed, I will do a new PR without.

Comment (catenacyber) [reply tests/test-bad-http-host-rule-1/test.yaml:8]: Would you please try without ?
This is a non-standard option that is rarely used ;-)

**Comment by jmtaylor90** [comment]
continued in https://github.com/OISF/suricata-verify/pull/886

**Pull request closed without merging**